File: rust/src/models/create_connected_account_request.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{connection_type::ConnectionType, customer_id::CustomerId};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CreateConnectedAccountRequest {
    pub connected_organization_id: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub connection_type: Option<ConnectionType>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub metadata: Option<serde_json::Value>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub platform_customer_id: Option<CustomerId>,
}

impl CreateConnectedAccountRequest {
    pub fn new(connected_organization_id: String) -> Self {
        Self {
            connected_organization_id,
            connection_type: None,
            metadata: None,
            platform_customer_id: None,
        }
    }
}

File: typescript/src/models/createConnectedAccountRequest.ts
// this file is @generated
import { type ConnectionType, ConnectionTypeSerializer } from "./connectionType";
import { type CustomerId, CustomerIdSerializer } from "./customerId";

export interface CreateConnectedAccountRequest {
  connectedOrganizationId: string;

  connectionType?: ConnectionType | null;

  metadata?: unknown;

  platformCustomerId?: CustomerId | null;
}

export const CreateConnectedAccountRequestSerializer = {
  _fromJsonObject(object: any): CreateConnectedAccountRequest {
    return {
      connectedOrganizationId: object["connected_organization_id"],
      connectionType:
        object["connection_type"] != null
          ? ConnectionTypeSerializer._fromJsonObject(object["connection_type"])
          : undefined,
      metadata: object["metadata"],
      platformCustomerId:
        object["platform_customer_id"] != null
          ? CustomerIdSerializer._fromJsonObject(object["platform_customer_id"])
          : undefined,
    };
  },

  _toJsonObject(self: CreateConnectedAccountRequest): any {
    return {
      connected_organization_id: self.connectedOrganizationId,
      connection_type:
        self.connectionType != null
          ? ConnectionTypeSerializer._toJsonObject(self.connectionType)
          : undefined,
      metadata: self.metadata,
      platform_customer_id:
        self.platformCustomerId != null
          ? CustomerIdSerializer._toJsonObject(self.platformCustomerId)
          : undefined,
    };
  },
};
